validator/src/command.rs

```diff
@@ -0,0 +1,254 @@
+
+    /// genesis hash
+    #[arg(short, long)]
+    genesis_hash: Option<String>,
+
+    /// Private key
+    #[arg(long)]
+    keypair_path: Option<String>,
+
+    /// Log Level
+    #[arg(short, long)]
+    log_level: Option<LevelFilter>,
+}
+
+#[derive(Args, Debug)]
+struct InitArgs {
+    /// rpc url
+    #[arg(short, long)]
+    rpc_url: String,
+
+    /// websocket url
+    #[arg(short, long)]
+    ws_url: String,
+
+    /// program ID
+    #[arg(long)]
+    program_id: String,
+
+    /// genesis hash
+    #[arg(short, long)]
+    genesis_hash: String,
+
+    /// Private key
+    #[arg(long)]
+    keypair_path: String,
+
+    /// Log Level
+    #[arg(short, long)]
+    log_level: Option<LevelFilter>,
+}
+
+#[derive(Clone, Debug)]
+pub enum Values {
+    Yes,
+    No,
+}
+
+impl FromStr for Values {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.eq_ignore_ascii_case("yes") {
+            Ok(Values::Yes)
+        } else if s.eq_ignore_ascii_case("no") {
+            Ok(Values::No)
+        } else {
+            Err(format!("Can not parse {}", s))
+        }
+    }
+}
+
+impl Display for Values {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+pub fn parse_config() -> Option<Config> {
+    let args = Cli::parse();
+    match args.command {
+        Commands::Run(cmd) => {
+            let config_file = config_file();
+            let config_data = fs::read_to_string(config_file).expect(
+                "Failed to read config file; make sure you’ve run init \
+                 command first.",
+            );
+            let default_config: Config = toml::from_str(&config_data).unwrap();
+            let keypair = if cmd.keypair_path.is_some() {
+                let keypair = read_keypair_file(&cmd.keypair_path.unwrap())
+                    .expect("Unable to read keypair file");
+                keypair.into()
+            } else {
+                default_config.keypair
+            };
+            let config = Config {
+                rpc_url: cmd.rpc_url.unwrap_or(default_config.rpc_url),
+                ws_url: cmd.ws_url.unwrap_or(default_config.ws_url),
+                program_id: cmd.program_id.unwrap_or(default_config.program_id),
+                genesis_hash: cmd
+                    .genesis_hash
+                    .unwrap_or(default_config.genesis_hash),
+                keypair,
+                log_level: cmd
+                    .log_level
+                    .unwrap_or(LevelFilter::Info)
+                    .to_string(),
+            };
+            setup_logging(LevelFilter::from_str(&config.log_level).unwrap());
+            Some(config)
+        }
+        Commands::Init(cmd) => {
+            let config_file = config_file();
+            let file = fs::metadata(config_file.clone());
+            setup_logging(cmd.log_level.unwrap_or(LevelFilter::Info));
+            if file.is_ok() {
```

Comment:
Actually it looks like all you care is checking whether the file exists.  In that case:
```suggestion
            if config_file.exists() {
```
and fs::metadata call isn’t necessary.